PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -2822,25 +2822,19 @@ void ScintillaEditView::currentLinesDown() const
 
 void ScintillaEditView::changeCase(__inout wchar_t * const strWToConvert, const int & nbChars, const TextCase & caseToConvert) const
 {
-	if (strWToConvert == nullptr || nbChars == NULL)
+	if (strWToConvert == nullptr || nbChars == 0)
 		return;
 
 	switch (caseToConvert)
 	{
 		case UPPERCASE:
 		{
-			for (int i = 0; i < nbChars; ++i)
-			{
-				strWToConvert[i] = (WCHAR)(UINT_PTR)::CharUpperW((LPWSTR)strWToConvert[i]);
-			}
+			::CharUpperBuffW(strWToConvert, nbChars);

```

Comment:
This works exactly the same as CharUpperW:
https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-charupperbuffa

Reply:
Though it could work exactly the same, we take the risk of regression, not even for bug-fixing or a new feature, but for removing the warning of a compiler that we don't use for releasing.

Reply:
I checked using commands from _Edit > Convert case to_ menu (to upper, to lower, inverting mixed case, etc). It works the same.
I thought, that in some cases calling a function on entire string rather than on each character individually could give a better result.
If you want, I can revert this part back to using `CharUpperW`.